packages/zipkin/test/batch-recorder.test.js

```diff
@@ -302,8 +302,35 @@ describe('Batch Recorder - integration test', () => {
         'http.status_code': '200'
       }
     });
+  });
+
+  it('should only flush spans when calling flush method', () => {
+    const clock = lolex.install();
+    recorder = new BatchRecorder({
+      logger: {
+        logSpan: (span) => {
+          spans.push(JSON.parse(JSON_V2.encode(span)));
+        }
+      }
+    });
+
+    recorder.record(record(rootId, now(), new Annotation.ServerRecv()));
 
     expect(spans).to.be.empty; // eslint-disable-line no-unused-expressions
+
+    clock.tick(100); // 1000 is de batching interval
```

Comment:
I would personally remove the lolex stuff as it is irrelevant. For example, we have other tests in the same file which show that records are not flushed until complete. The only responsibility this one should have is that when you hit flush the data is expected. The clock ticking stuff distracts from an otherwise simple goal IOTW. For this test you can even use fake timestamps like other things here

Reply:
Right. I think for me it was better to make it clear that the flush occurs when it was not yet the time for a scheduled flush.